go-controller/pkg/cluster/ha_master.go

```diff
@@ -0,0 +1,321 @@
+		resourcelock.ConfigMapsResourceLock,
+		config.Kubernetes.OVNConfigNamespace,
+		haLeaderLockName,
+		hacluster.kubeClient.CoreV1(),
+		nil,
+		resourcelock.ResourceLockConfig{
+			Identity:      hacluster.nodeName,
+			EventRecorder: nil,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	HAClusterNewLeader := func(identity string) {
+		logrus.Infof("The new leader is : " + identity)
+		wasLeader := hacluster.isLeader
+		if hacluster.nodeName == identity {
+			err = hacluster.ConfigureAsActive(identity)
+			if err != nil {
+				logrus.Errorf(err.Error())
+				panic(err.Error())
+			}
+			hacluster.isLeader = true
+		} else {
+			if wasLeader {
+				//This node was leader and it lost the election.
+				// Whenever the node transitions from leader to follower,
+				// we need to handle the transition properly like clearing
+				// the cache. It is better to exit for now.
+				// kube will restart and this will become a follower.
+				logrus.Infof("[%s] was the leader and it lost the election. Exiting", hacluster.nodeName)
+				os.Exit(1)
+			}
+
+			err = hacluster.ConfigureAsStandby(identity)
+			if err != nil {
+				logrus.Errorf(err.Error())
+				panic(err.Error())
+			}
+
+		}
+	}
+
+	lec := leaderelection.LeaderElectionConfig{
+		Lock:          rl,
+		LeaseDuration: leaseDuration,
+		RenewDeadline: renewDeadline,
+		RetryPeriod:   retryPeriod,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {},
+			OnStoppedLeading: func() {
+				//This node was leader and it lost the election.
+				// Whenever the node transitions from leader to follower,
+				// we need to handle the transition properly like clearing
+				// the cache. It is better to exit for now.
+				// kube will restart and this will become a follower.
+				logrus.Infof("[%s] stopped Leading. Exiting", hacluster.nodeName)
+				os.Exit(1)
+			},
+			OnNewLeader: HAClusterNewLeader,
+		},
+	}
+
+	go leaderelection.RunOrDie(context.Background(), lec)
+
+	return nil
+}
+
+// ConfigureAsActive configures the node as active.
+func (hacluster *OvnHAMasterController) ConfigureAsActive(masterNodeName string) error {
+	if hacluster.manageDBServers {
+		// Find the endpoint for the service
+		ep, err := hacluster.cluster.Kube.GetEndpoint(config.Kubernetes.OVNConfigNamespace, ovnkubeDbEp)
+		createEp := true
+		if err == nil {
+			if ep.Name == ovnkubeDbEp {
+				createEp = false
+			}
+		}
+
+		const (
+			nbPort int32 = 6641
+			sbPort int32 = 6642
+		)
+
+		ovndbEp := v1.Endpoints{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: config.Kubernetes.OVNConfigNamespace,
+				Name:      ovnkubeDbEp,
+			},
+			Subsets: []v1.EndpointSubset{
+				{
+					Addresses: []v1.EndpointAddress{
+						{IP: config.Kubernetes.OVNKubePodIP},
+					},
+					Ports: []v1.EndpointPort{
+						{
+							Name: "north",
+							Port: nbPort,
+						},
+						{
+							Name: "south",
+							Port: sbPort,
+						},
+					},
+				},
+			},
+		}
+
+		if createEp {
```

Comment:
The usual convention is ot just always call `Update()`, and if the error code is `apierrors.IsNotFound(err)`, then do `Create()`